tests/core/resource/test_views.py

```diff
@@ -485,11 +485,16 @@ def test_invalid_content_type_header_on_record_returns_415(self):
                                    status=415)
         self.assertEqual(resp.json['code'], 415)
         messages = (
-            "Content-Type header should be one of " +
-            "['application/merge-patch+json', 'application/json']",
-            "Content-Type header should be one of " +
-            "['application/json', 'application/merge-patch+json']")
-        self.assertIn(resp.json['message'], messages)
+            "Content-Type header should be one of [",
+            "'application/json-patch+json'",
+            ", ",
+            "'application/json'",
+            ", ",
+            "'application/merge-patch+json'",
+            "]")
+        for message in messages:
+            self.assertIn(message, resp.json['message'])
+        self.assertEquals(len("".join(messages)), len(resp.json['message']))
```

Comment:
I think this is OK. It's pretty clever and probably the best you can do without using something like `eval`. Of course, you don't need `", "` twice -- you might take it out entirely just to make it clear that we're not comparing against the literal text of the error message. How about something like this?

```python
self.assert_(resp.json['message'].startswith('Content-Type header should be one of ['))
self.assert_(resp.json['message'].endswith(']'))
for message in messages:
    self.assertIn(message, resp.json['message'])
```